.github/workflows/workflow.yml

```diff
@@ -135,6 +135,51 @@ jobs:
             --access Allow \
             --protocol Tcp \
             --destination-port-ranges 5000 && \
+            az network nsg rule create \
+            --resource-group "${{ env.STAGE }}-${{ env.NAME }}-${{ env.NEW_COLOUR }}" \
+            --nsg-name "${{ env.STAGE }}-${{ env.NAME }}-${{ env.NEW_COLOUR }}NSG" \
+            --name AllowPrometheusPort9157 \
+            --priority 1011 \
+            --access Allow \
+            --protocol Tcp \
+            --destination-port-ranges 9157 && \
+            az vm run-command invoke \
+            --resource-group "${{ env.STAGE }}-${{ env.NAME }}-${{ env.NEW_COLOUR }}" \
+            --name "${{ env.STAGE }}-${{ env.NAME }}-${{ env.NEW_COLOUR }}" \
+            --command-id RunShellScript \
+            --scripts "\
+              set -eux
+              adduser prometheus-client --disabled-password --gecos ''
+              cd /home/prometheus-client/
+              PROMETHEUS_VERSION=1.7.0
+              wget https://github.com/prometheus/node_exporter/releases/download/v\$PROMETHEUS_VERSION/node_exporter-\$PROMETHEUS_VERSION.linux-amd64.tar.gz
+              tar -xvzf node_exporter-\$PROMETHEUS_VERSION.linux-amd64.tar.gz
+              echo \"\
+            [Unit]
+            Description=Prometheus Node Exporter
+            Wants=network-online.target
+            After=network-online.target
+
+            [Service]
+            User=prometheus-client
+            Group=prometheus-client
+            Type=simple
+            ExecStart=/home/prometheus-client/node_exporter-\$PROMETHEUS_VERSION.linux-amd64/node_exporter \\\\
+                --collector.systemd \\\\
+                --web.listen-address=:9157 \\\\
+                --web.config.file /home/prometheus-client/web-config.yaml \\\\
+                --collector.textfile.directory /home/prometheus-client/textfile_directory
```

Comment:
This line can be canned; textfile_directory won't actually exist anyway and there's nothing on the server that puts things into it for now (and I wouldn't expect there to be soon).